src/experiments/75.shader-matcap-transition.tsx

```diff
@@ -0,0 +1,248 @@
+        vec3 c = vec3(1.0, 1.0, 1.0);
+        vec3 d = vec3(0.00, 0.33, 0.67);
+  
+        vec3 viewDir = normalize(vView); 
+        vec3 x = normalize(vec3(viewDir.z, 0.0, -viewDir.x));
+        vec3 y = cross(viewDir, x);
+        vec2 uv = vec2(dot(x, vNormal), dot(y, vNormal)) * 0.495 + 0.5;
+  
+        vec4 color = texture2D(uMatcap, uv);
+        gl_FragColor = vec4(vNormal, 1.0); 
+  
+        float diff = abs(dot(vNormal, normalize(vec3(1.0, 1.0, 0.0)))) + abs(dot(vNormal, normalize(vec3(1.0, -1.0, 0.0))));
+        diff *= 0.5;
+  
+       
+     
+        
+        float noise = 0.5 * (cnoise(vPosition) + 1.0) + uProgress - vPosition.y * 0.08;
+        float step = smoothstep(0.1, 0.09,noise);
+        vec3 animateColor = a + b * cos(2.0 * 3.1415 * (c * diff + d + uTime / 3.0));
+        
+        vec4 final = mix(color, vec4(animateColor, 1.0), step);
+  
+        gl_FragColor = final; 
+      }
+  `
+interface ExtendedScrollControlsState extends ScrollControlsState {
+  scroll: {
+    current: number
+  }
+}
+
+interface ScrollControlsState {
+  scroll: {
+    current: number
+  }
+}
+
+const Experience = () => {
+  const matcap = useTexture('/matcaps/darkblue.png')
+  const pyramidGeometry = useMemo(
+    () => new THREE.CylinderGeometry(0, 4, 5, 4, 1),
+    []
+  )
+  const pyramidRef = useRef<THREE.Mesh>(null)
+  const cameraRef = useRef<THREE.PerspectiveCamera>(null)
+  const data: ExtendedScrollControlsState =
+    useScroll() as unknown as ExtendedScrollControlsState
+
+  const customMaterial = useMemo(
+    () =>
+      new THREE.ShaderMaterial({
+        side: THREE.DoubleSide,
+        uniforms: {
+          uTime: { value: 0 },
+          uMatcap: { value: matcap },
+          uProgress: { value: 0 }
+        },
+        transparent: true,
+        vertexShader: vertex,
+        fragmentShader: fragment
+      }),
+    [matcap]
+  )
+
+  useFrame((_state, delta) => {
+    customMaterial.uniforms.uProgress.value = -data.scroll.current
+    customMaterial.uniforms.uTime.value += delta
+    cameraRef.current?.position.set(0, 3, 10 - data.scroll.current * 6)
+    cameraRef.current?.lookAt(0, 0, 0)
+    const rotationAngle = data.scroll.current * Math.PI
+    pyramidRef.current?.rotation.set(0, rotationAngle, 0)
+  })
+
+  return (
+    <>
+      <PerspectiveCamera ref={cameraRef} makeDefault />
+      <mesh
+        ref={pyramidRef}
+        geometry={pyramidGeometry}
+        material={customMaterial}
+        position={[0, 1, 0]}
+        rotation={[0, 0, 0]}
+        scale={0.4}
+      >
+        <Edges linewidth={2} scale={1} threshold={15} color="white" />
+      </mesh>
+      <Grid
+        position={[0, -0.01, 0]}
+        args={[10.5, 10.5]}
+        cellSize={0.6}
+        cellThickness={1.0}
+        cellColor={'#727272'}
+        sectionSize={3.3}
+        sectionThickness={1.5}
+        sectionColor={'#8d8d8d'}
+        fadeDistance={25}
+        fadeStrength={1.0}
+        infiniteGrid
```

Comment:
Should we make the strong lines match the sublte ones?


![image](https://github.com/user-attachments/assets/4b871a6e-025d-4672-9216-70a88ef2e7b8)

The grid size could be setted with round numbers and then use the scale prop to adjust it:

```tsx
<Grid
  scale={[0.6, 0.6, 0.6]}
  position={[0, -0.01, 0]}
  cellSize={1}
  sectionSize={5}
  ... other props
/>
```

![image](https://github.com/user-attachments/assets/b57f2d46-2c0c-44e0-a68c-d7fea9b02fdb)